Microsoft.Toolkit.Uwp.UI.Controls.Input/ColorPicker/ColorPicker.Properties.cs

```diff
@@ -20,7 +20,7 @@ public partial class ColorPicker
                 nameof(CustomPaletteColors),
                 typeof(ObservableCollection<Windows.UI.Color>),
                 typeof(ColorPicker),
-                new PropertyMetadata(Windows.UI.Color.FromArgb(0x00, 0x00, 0x00, 0x00)));
+                new PropertyMetadata(DependencyProperty.UnsetValue));
```

Comment:
We should **never** be using UnsetValue here, this should be always be `null`, if not a method. We should only use `UnsetValue` for value converters (bug #3311).

See docs for [`PropertyMetadata`](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.PropertyMetadata#remarks) and [`UnsetValue`](https://docs.microsoft.com/uwp/api/windows.ui.xaml.dependencyproperty.unsetvalue#remarks) both call this out.

Can you fix this here an in the other places as well? Thanks!

Reply:
@michael-hawker Ok, I will update. This was my misunderstanding. It would make sense that the property system supported this case but if it doesn't, it doesn't.

Reply:
Yeah, no worries. Sorry we missed it in the initial PR. It's extra confusing because it's the opposite guidance for the Value Converters. We just haven't had a chance to go fix those yet with all the other work we did for 7.0. Thanks a bunch for working through the fix on the issue thread with the original reporter! 🦙❤

Reply:
I noticed some other occurrences of UnsetValue doing a quick search. Here is an example:

https://github.com/windows-toolkit/WindowsCommunityToolkit/blob/68888d2d7f0eba1b1f671a7ba1acc775ac0a46ef/Microsoft.Toolkit.Uwp.UI/Extensions/UIElementExtensions.cs#L19-L23

I think the code base needs to be audited for this situation a bit more.

Reply:
I've made the changes and updated the converters I added to return UnsetValue instead of throwing exceptions.

I still can't believe it isn't supported in registration like this. DependencyObject.ReadLocalValue may return UnsetValue. ClearValue() uses it I think? I would also expect the default PropertyMetadata to use UnsetValue instead of null. Oh well, it's good to be aware of this.